src/StreamRequest.jl

```diff
@@ -103,9 +103,13 @@ end
 writechunk(stream, body::IO) = writebodystream(stream, body)
 writechunk(stream, body) = write(stream, body)
 
-function readbody(stream::Stream, res::Response, redirectlimitreached)
+function readbody(stream::Stream, res::Response, redirectlimitreached, decompress)
     if isbytes(res.body)
-        res.body = read(stream)
+        if decompress && header(res, "Content-Encoding") == "gzip"
+            res.body = transcode(GzipDecompressor, read(stream))
```

Comment:
Is there a streaming version of this that does this in one go?

Reply:
Good point; there is. And that actually simplifies it because we can just wrap the `stream` in the decompressor and get the decompressing for both static byte vector and streaming response case.